### PR TITLE
release: 2024-09-02

### PR DESCRIPTION
**Pull request by cb22**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,113 @@
...
+
+- [#2265](https://github.com/tigerbeetle/tigerbeetle/pull/2265)
+
+  Don't cancel in-progress GitHub actions on the main branch. In particular, this ensures that the
+  devhub records the benchmark measurements for every merge to main, even if those merges occur in
+  quick succession.
+
+### TigerTracks 🎧
+
+- [I Want To Break Free](https://open.spotify.com/track/7iAqvWLgZzXvH38lA06QZg?si=a5ad69b31f3a45dd)
```
Comment (matklad) [inline CHANGELOG.md]: This is soooo perfect <3

File: CHANGELOG.md [hunk 2]
```diff
@@ -3,6 +3,113 @@
 Subscribe to the [tracking issue #2231](https://github.com/tigerbeetle/tigerbeetle/issues/2231)
 to receive notifications about breaking changes!
 
+## TigerBeetle 0.16.0
```
Comment (matklad) [inline CHANGELOG.md:6]: As per our new release process, you should merge new additions with the past unreleased entries:

https://github.com/tigerbeetle/tigerbeetle/blob/main/docs/about/internals/releases.md#skipping-release

We don't want require our users to piece together the actual changelog!

File: CHANGELOG.md [hunk 3]
```diff
@@ -3,6 +3,113 @@
...
+
+If you're running a client older than 0.16.0, you'll see the old behavior where zero amount
+transfers are disallowed, but on newer clients these are supported and will create a transfer with
+an amount of 0.
+
+Additionally, the sentinel value to representing posting the full amount of a pending transfer, or
+doing a balancing transfer for as much as possible has changed. It's no longer 0, but instead
+`AMOUNT_MAX`.
+
+See the [**tracking issue**](https://github.com/tigerbeetle/tigerbeetle/issues/2231#issuecomment-2305132591) for more details.
```
Comment (matklad) [inline CHANGELOG.md:22]: :+1: perfect use of lead paragraph to explain _why_ we bump the version!

File: CHANGELOG.md [hunk 4]
```diff
@@ -3,6 +3,113 @@
...
+
+### Safety And Performance
+
+- [#2255](https://github.com/tigerbeetle/tigerbeetle/pull/2255)
+
+  The new state sync protocol regressed the behavior where the replica would try to repair the WAL
+  before switching to state sync, and this puts the old behavior back in.
+
+  [WAL repair](https://docs.tigerbeetle.com/about/internals/vsr#protocol-repair-wal) is used when the lagging replica's log still intersects with the cluster's
+  current log, while [state sync](https://docs.tigerbeetle.com/about/internals/sync) is used when when the logs no longer intersect.
```
Comment (matklad) [inline CHANGELOG.md]: Try to wrap at 100 by putting URLs on lines of their own

File: CHANGELOG.md [hunk 5]
```diff
@@ -3,6 +3,113 @@
...
+
+- [#2244](https://github.com/tigerbeetle/tigerbeetle/pull/2244)
+  Try to repair (but not commit) prepares, even if we don't have all the headers between checkpoint
```
Comment (matklad) [inline CHANGELOG.md]:
```suggestion
- [#2244](https://github.com/tigerbeetle/tigerbeetle/pull/2244)

  Try to repair (but not commit) prepares, even if we don't have all the headers between checkpoint
```

File: CHANGELOG.md [hunk 6]
```diff
@@ -3,6 +3,113 @@
...
+  number for a release be?). Instead, make it so this very changelog is the source of truth for
+  the version number which is explicitly set.
+
+- [#2248](https://github.com/tigerbeetle/tigerbeetle/pull/2248)
+
+  Inside the VOPR, heal both WAL header sectors before replica startup.
+
+  Before, the logic would arbitrarily repair one WAL header sectors if both sectors are corrupt,
+  which could cause seeds to fail when only one one of the WAL header sectors is corrupt and the
+  other one is corrupted on startup.
```
Comment (matklad) [inline CHANGELOG.md]: Feels inconsequential enough to cut from relnotes. We don't want to put _every_ PR into relnotes.

Not a big deal either way!

File: CHANGELOG.md [hunk 7]
```diff
@@ -3,6 +3,113 @@
...
+
+- [#2252](https://github.com/tigerbeetle/tigerbeetle/pull/2252)
+
+  Change `init` function signatures to allow for in-place initialization. This addresses the silent
+  stack growth caused by intermediate copy/move allocations during the initialization of large
+  objects.
+
+  Specifically, the `Forest` struct can grow indefinitely depending on the number of
+  `Grooves`/`IndexTrees` needed to support the StateMachine's custom logic, causing TigerBeetle to
+  crash during startup due to stack-overflow.
```
Comment (matklad) [inline CHANGELOG.md:130]: In contrast, this one is well-worth including, as it illuminates a very interesting aspect of Zig, even though it's an internal change. 

I guess I mentally use the following criteria to decide between inclusion and truncation of a PR:

* Is there a category of readers which would be glad that the PR is included?

File: CHANGELOG.md (hunk 2 above)
Comment (cb22) [reply CHANGELOG.md:6]: Makes sense - I missed that last sentence when checking out the diff!

**Review by matklad — APPROVED** [review]

**Pull request merged**